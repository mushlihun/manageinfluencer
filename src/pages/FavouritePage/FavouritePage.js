import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as HeartIcon } from '../../icons/heart.svg'

export default function FavouritePage({ influencerData, onFavouriteClick }) {
  return (
    <section className="profiles">
      {influencerData?.map((profile) => {
        return (
          <WrapperStyled key={profile.id}>
            <ImageStyled src={profile.image} />
            <UsernameStyled>{profile.username}</UsernameStyled>
            <CategoriesStyled>{profile.categories.join(', ')}</CategoriesStyled>
            <DetailListStyled>
              <DetailItemStyled key={profile.username}>
                Follower: {profile.follower.toLocaleString('en-US')}
              </DetailItemStyled>
              <DetailItemStyled key={profile.gender}>
                Gender: {profile.gender}
              </DetailItemStyled>
              <DetailItemStyled key={profile.age}>
                Age: {profile.age}
              </DetailItemStyled>
              <DetailItemStyled key={profile.location}>
                Location: {profile.location}
              </DetailItemStyled>
              <DetailItemStyled key={profile.languages}>
                Languages: {profile.languages.join(', ')}
              </DetailItemStyled>
            </DetailListStyled>
            <ButtonWrapperStyled>
              <FavouriteButtonStyled
                id="like-button"
                onClick={() => onFavouriteClick(profile)}
                isFavourite={profile.isFavourite}
              >
                <HeartStyled id="like-button" fill={'var(--darkpink)'} />
              </FavouriteButtonStyled>
              <AnchorStyled href="mailto:influencer@gmail.com">
                <ButtonStyled id="contact-button">Contact</ButtonStyled>
              </AnchorStyled>
            </ButtonWrapperStyled>
          </WrapperStyled>
        )
      })}
    </section>
  )
}

const WrapperStyled = styled.section`
  border: 1px solid white;
  border-radius: 15px 50px;
  margin: 10px 20px 30px 20px;
`

const ButtonWrapperStyled = styled.section`
  text-align: center;
`

const FavouriteButtonStyled = styled.button`
  background: var(--lightpink);
  border-radius: 50%;
  text-align: center;
  border: none;
  stroke: #e693b5;
  cursor: pointer;
  height: 70px;
  width: 70px;
  margin: 20px;
`

const HeartStyled = styled(HeartIcon)``

const ButtonStyled = styled.button`
  color: var(--main-font-color);
  font-family: var(--font-weight-light);
  background: var(--darkgreen);
  border: var(--darkgreen);
  box-shadow: 0 3px 10px var(--boxshadow-color);
  font-size: 18px;
  border-radius: 40px;
  width: 120px;
  height: 75px;
  cursor: pointer;
  margin: 20px;
`

const ImageStyled = styled.img`
  width: 120px;
  height: 190px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
`

const UsernameStyled = styled.h2`
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  font-size: 20px;
  text-align: center;
  padding: 10px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
`
const CategoriesStyled = styled.h3`
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 18px;
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
`

const DetailListStyled = styled.ul`
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  width: 300px;
  margin-left: auto;
  margin-right: auto;
`
const DetailItemStyled = styled.li`
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
`
const AnchorStyled = styled.a`
  text-decoration: none;
`

FavouritePage.propTypes = {
  influencerData: PropTypes.array,
}
