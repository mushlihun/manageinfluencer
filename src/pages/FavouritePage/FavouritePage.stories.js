import React from 'react'
import FavouritePage from './FavouritePage'
import influencerData from '../../mocks/influencer.json'
import HeaderFavourites from '../../components/Header/HeaderFavourites'

export default {
  title: 'Pages/FavouritePage',
  component: FavouritePage,
}

export const fullList = () => {
  return (
    <>
      {' '}
      <HeaderFavourites headline="Favourites" />{' '}
      <FavouritePage influencerData={influencerData} />{' '}
    </>
  )
}
