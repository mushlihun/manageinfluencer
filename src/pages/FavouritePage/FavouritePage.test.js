import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import FavouritePage from './FavouritePage.js'
import 'jest-styled-components'

describe('FavouritePage', () => {
  const influencerData = [
    {
      id: 1,
      username: '@fitness4life',
      categories: ['Fitness'],
      follower: '10k',
      gender: 'female',
      age: '25 years',
      location: 'Hamburg',
      languages: ['German, ', 'English'],
    },
    {
      id: 2,
      username: '@VeganPlace',
      categories: ['Food', 'Health'],
      follower: '1.000',
      gender: 'female',
      age: '28 years',
      location: 'Munich',
      languages: ['German, ', 'English'],
    },
  ]

  it('displays the information details of an influencer', () => {
    const { getByText, getAllByText } = render(
      <FavouritePage influencerData={influencerData} />
    )
    influencerData.forEach(
      ({
        username,
        follower,
        location,
        age,
        gender,
        languages,
        categories,
      }) => {
        expect(getByText(username)).toBeInTheDocument()
        expect(getByText(new RegExp(follower, 'i'))).toBeInTheDocument()
        expect(getByText(new RegExp(age, 'i'))).toBeInTheDocument()
        expect(getByText(new RegExp(location, 'i'))).toBeInTheDocument()
        expect(getAllByText(new RegExp(gender, 'i'))[0]).toBeInTheDocument()
        expect(JSON.stringify(new RegExp(languages, 'i'))).toBe(
          JSON.stringify(/languages/i)
        )
        expect(JSON.stringify(new RegExp(categories, 'i'))).toBe(
          JSON.stringify(/categories/i)
        )
      }
    )
  })

  it('renders favourites details correctly', () => {
    const tree = renderer.create(
      <FavouritePage influencerData={influencerData} />
    )
    expect(tree).toMatchSnapshot()
  })
})
