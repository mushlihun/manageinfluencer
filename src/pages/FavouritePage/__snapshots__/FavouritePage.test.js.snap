// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritePage renders favourites details correctly 1`] = `
.c0 {
  border: 1px solid white;
  border-radius: 15px 50px;
  margin: 10px 20px 30px 20px;
}

.c6 {
  text-align: center;
}

.c7 {
  background: var(--lightpink);
  border-radius: 50%;
  text-align: center;
  border: none;
  stroke: #e693b5;
  cursor: pointer;
  height: 70px;
  width: 70px;
  margin: 20px;
}

.c9 {
  color: var(--main-font-color);
  font-family: var(--font-weight-light);
  background: var(--darkgreen);
  border: var(--darkgreen);
  box-shadow: 0 3px 10px var(--boxshadow-color);
  font-size: 18px;
  border-radius: 40px;
  width: 120px;
  height: 75px;
  cursor: pointer;
  margin: 20px;
}

.c1 {
  width: 120px;
  height: 190px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
}

.c2 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  font-size: 20px;
  text-align: center;
  padding: 10px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
}

.c3 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 18px;
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c4 {
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  width: 300px;
  margin-left: auto;
  margin-right: auto;
}

.c5 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<section
  className="profiles"
>
  <section
    className="c0"
  >
    <img
      className="c1"
    />
    <h2
      className="c2"
    >
      @fitness4life
    </h2>
    <h3
      className="c3"
    >
      Fitness
    </h3>
    <ul
      className="c4"
    >
      <li
        className="c5"
      >
        Follower: 
        10k
      </li>
      <li
        className="c5"
      >
        Gender: 
        female
      </li>
      <li
        className="c5"
      >
        Age: 
        25 years
      </li>
      <li
        className="c5"
      >
        Location: 
        Hamburg
      </li>
      <li
        className="c5"
      >
        Languages: 
        German, , English
      </li>
    </ul>
    <section
      className="c6"
    >
      <button
        className="c7"
        id="like-button"
        onClick={[Function]}
      >
        <svg
          className=""
          fill="var(--darkpink)"
          id="like-button"
        >
          heart.svg
        </svg>
      </button>
      <a
        className="c8"
        href="mailto:influencer@gmail.com"
      >
        <button
          className="c9"
          id="contact-button"
        >
          Contact
        </button>
      </a>
    </section>
  </section>
  <section
    className="c0"
  >
    <img
      className="c1"
    />
    <h2
      className="c2"
    >
      @VeganPlace
    </h2>
    <h3
      className="c3"
    >
      Food, Health
    </h3>
    <ul
      className="c4"
    >
      <li
        className="c5"
      >
        Follower: 
        1.000
      </li>
      <li
        className="c5"
      >
        Gender: 
        female
      </li>
      <li
        className="c5"
      >
        Age: 
        28 years
      </li>
      <li
        className="c5"
      >
        Location: 
        Munich
      </li>
      <li
        className="c5"
      >
        Languages: 
        German, , English
      </li>
    </ul>
    <section
      className="c6"
    >
      <button
        className="c7"
        id="like-button"
        onClick={[Function]}
      >
        <svg
          className=""
          fill="var(--darkpink)"
          id="like-button"
        >
          heart.svg
        </svg>
      </button>
      <a
        className="c8"
        href="mailto:influencer@gmail.com"
      >
        <button
          className="c9"
          id="contact-button"
        >
          Contact
        </button>
      </a>
    </section>
  </section>
</section>
`;
